src/Microsoft.Azure.ServiceBus/Amqp/AmqpLinkCreator.cs

```diff
@@ -46,19 +46,33 @@ protected AmqpLinkCreator(string entityPath, ServiceBusConnection serviceBusConn
             AmqpSession session = null;
             try
             {
-                // Create our Session
+                // Create Session
                 AmqpSessionSettings sessionSettings = new AmqpSessionSettings { Properties = new Fields() };
                 session = connection.CreateSession(sessionSettings);
                 await session.OpenAsync(timeoutHelper.RemainingTime()).ConfigureAwait(false);
+            }
+            catch (Exception exception)
+            {
+                MessagingEventSource.Log.AmqpSessionCreationException(this.entityPath, connection, exception);
+                session?.Abort();
+                throw;
+            }
 
-                // Create our Link
+            try
+            {
+                // Create Link
                 AmqpObject link = this.OnCreateAmqpLink(connection, this.amqpLinkSettings, session);
                 await link.OpenAsync(timeoutHelper.RemainingTime()).ConfigureAwait(false);
                 return link;
             }
-            catch (Exception)
+            catch (Exception exception)
             {
-                session?.Abort();
```

Comment:
I have not removed it. Just seperated the session creation/open from the link creation and not aborting the session when link creation fails